### PR TITLE
add legend as dict for line plots with test

### PR DESCRIPTION
**Pull request by gidden**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Adding to RELEASE_NOTES.md

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

File: pyam/plotting.py [hunk 1]
```diff
@@ -472,9 +475,10 @@ def line_plot(df, x='year', y='value', ax=None, legend=None, title=True,
         The column to use for y-axis values
         default: value
     ax : matplotlib.Axes, optional
-    legend : bool, optional
-        Include a legend (`None` displays legend only if less than 13 entries)
-        default: None
+    legend : bool or dictionary, optional
+        Add a legend. If a dictionary is provided, it will be used as keyword
+        arguments in creating the legend.
+        default: None (displays legend only if less than 13 entries)
```
Comment (danielhuppmann) [inline pyam/plotting.py]: rephrase to 'as specified by `MAX_LEGEND_LABELS`'?

**Comment by gidden** [comment]
Perhaps, if a user reads the docstring, they won't know what value
MAX_LEGEND_LABELS is. In the future, we should make that configurable (much
like matplotlib has rcparams), but I'm not sure that time is now..

**Comment by gidden** [comment]
ok @danielhuppmann, this now implements a more intuitive way of setting our labels and I can now confirm that lineplot by composition works as I expect it to! Just tested with alphas.

**Comment by gidden** [comment]
@danielhuppmann did you mean to add a commit here?

**Comment by danielhuppmann** [comment]
@gidden just resolved the merge conflict from accepting PR #70 so that I can merge

**Comment by gidden** [comment]
great, thanks!

**Pull request merged**